gwvolman/tasks.py

```diff
@@ -708,3 +653,173 @@ def rebuild_image_cache():
+
+    run = self.girder_client.get('/run/{}'.format(run_id))
+    tale = self.girder_client.get('/tale/{}'.format(tale_id))
+    user = self.girder_client.get('/user/{}'.format(user_id))
+
+    self.job_manager.updateProgress(
+        message='Preparing volumes', total=RECORDED_RUN_STEP_TOTAL,
+        current=1, forceFlush=True)
+
+    # Create Docker volume
+    vol_name = "%s_%s_%s" % (run_id, user['login'], new_user(6))
+    mountpoint = _create_docker_volume(cli, vol_name)
+
+    # Create fuse directories
+    _make_fuse_dirs(mountpoint, ['data', 'workspace'])
+
+    # Mount data and workspace for the run
+    api_key = _get_api_key(self.girder_client)
+    session = _get_session(self.girder_client, version_id=run['runVersionId'])
+    if session['_id'] is not None:
+        _mount_girderfs(mountpoint, 'data', 'wt_dms', session['_id'], api_key)
+
+    _mount_girderfs(mountpoint, 'workspace', 'wt_run', run_id, api_key)
+
+    # Build the image for the run
+    self.job_manager.updateProgress(
+        message='Building image', total=RECORDED_RUN_STEP_TOTAL,
+        current=2, forceFlush=True)
+
+    # Setup image tag
+    build_time = int(time.time())
+    tag = '{}/{}/{}'.format(urlparse(DEPLOYMENT.registry_url).netloc,
+                            run_id, str(build_time))
+
+    # TODO: Need the environment.json in the workspace. Could be done via fuse.
+    work_dir = os.path.join(mountpoint, 'workspace')
+    image = self.girder_client.get('/image/%s' % tale['imageId'])
+    print("Dumping the environment to " + work_dir)
+    with open(os.path.join(HOSTDIR + work_dir, 'environment.json'), 'w') as fp:
+        json.dump(image, fp)
+
+    # TODO: What should we use here? Latest? What the tale was built with?
+    repo2docker_version = REPO2DOCKER_VERSION
+    print(f"Using repo2docker {repo2docker_version}")
+
+    # Build currently assumes tmp directory, in this case mount the run workspace
+    container_config = _get_container_config(self.girder_client, tale)
+    work_target = os.path.join(container_config.target_mount, 'workspace')
+    extra_volume =  {
+        work_dir: {
+            'bind': work_target,
+            'mode': 'rw'
+        }
+    }
+
+    try:
+        print("Building mage for recorded run " + tag)
+        ret = _build_image(cli, tale_id, image, tag, work_target, repo2docker_version, extra_volume)
+        if ret['StatusCode'] != 0:
+            raise ValueError('Image build failed for recorded run {}'.format(run_id))
+        # TODO: Delete the environment.json?
+        # TODO: Do we push the image? Delete it at the end?
+
+        self.job_manager.updateProgress(
+            message='Recording run', total=RECORDED_RUN_STEP_TOTAL,
+            current=3, forceFlush=True)
+
+        _recorded_run(cli, mountpoint, container_config, tag)
+
+        self.job_manager.updateProgress(
+            message='Finished recorded run', total=RECORDED_RUN_STEP_TOTAL,
+            current=4, forceFlush=True)
+
```

Comment:
does `try:` without `except:` actually work?

Reply:
The finally will execute, but obviously would be better with except.